opnode/rollup/derive/invert.go

```diff
@@ -1,26 +1,58 @@
 package derive
 
 import (
-	"encoding/binary"
+	"errors"
 	"fmt"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
+// setL1BlockValues(uint256 _number, uint256 _timestamp, uint256 _basefee, bytes32 _hash)
+type L1BlockInfo struct {
+	Number    uint64
+	Time      uint64
+	BaseFee   *big.Int
+	BlockHash common.Hash
+}
+
 // L1InfoDepositTxData is the inverse of L1InfoDeposit, to see where the L2 chain is derived from
 func L1InfoDepositTxData(data []byte) (nr uint64, time uint64, baseFee *big.Int, blockHash common.Hash, err error) {
-	if len(data) != 4+8+8+32+32 {
-		err = fmt.Errorf("data is unexpected length: %d", len(data))
-		return
+	info, err := L1InfoDepositTxDataToStruct(data)
+	return info.Number, info.Time, info.BaseFee, info.BlockHash, err
+}
+
+// L1InfoDepositTxDataToStruct is the inverse of L1InfoDeposit, to see where the L2 chain is derived from
+func L1InfoDepositTxDataToStruct(data []byte) (L1BlockInfo, error) {
+	out := L1BlockInfo{}
+	if len(data) != 4+32+32+32+32 {
+		return out, fmt.Errorf("data is unexpected length: %d", len(data))
+	}
+
+	// Number
+	offset := 4 // Selector hash. Should check
+	number := new(big.Int)
+	number.SetBytes(data[offset : offset+32])
+	if !number.IsUint64() {
```

Comment:
A malicious sequencer can cause this to return an error, what are the implications of that?

Reply:
Before we get to this codepath, we would have already checked the signature on the unsafe block received via p2p. So the worst thing that could happen is a malicious sequencer halts unsafe block distribution and then the verifier would have to wait until the transactions are batch submitted. So the protection against this is governance voting the malicious sequencer out. A sequencer would not want to do this because they want economic activity on their chain and doing this would result in degraded experience for users. Does that sound correct to you as a security model?

Reply:
Yep. I guess at the same time the sequencer could be building the chain at the same time.
It's probably worth fixing this at the ABI level, but I didn't quite get around to it in this PR- I think we need a more comprehensive look at this as well.